crates/core/src/syn/parser/value.rs

```diff
@@ -66,23 +68,55 @@ impl Parser<'_> {
 			}
 			t!("{") => {
 				self.pop_peek();
-				let object = self.parse_value_object::<SurrealQL>(stk, token.span).await?;
-				//HACK: This is an annoying hack to have geometries work.
-				//
-				// Geometries look exactly like objects and are a strict subsect of objects.
-				// However in code they are distinct and in surrealql the have different
-				// behavior.
-				//
-				// Geom functions don't work with objects and vice-versa.
-				//
-				// The previous parse automatically converted an object to geometry if it found
-				// an matching object. Now it no longer does that and relies on the
-				// 'planning' stage to convert it. But here we still need to do it in the
-				// parser.
-				if let Some(geom) = PublicGeometry::try_from_object(&object) {
-					PublicValue::Geometry(geom)
-				} else {
-					PublicValue::Object(object)
+
+				if self.eat(t!("}")) {
+					return Ok(PublicValue::Object(PublicObject::new()));
+				}
+
+				// First, check if it's an empty set. `{,}` is an empty set.
+				if self.eat(t!(",")) && self.eat(t!("}")) {
+					return Ok(PublicValue::Set(PublicSet::new()));
+				}
+
+				// Use glue_and_peek1 for simple tokens
+				match self.glue_and_peek1()?.kind {
+					t!(":") => {
+						// It's an object
+						let object = self.parse_value_object::<SurrealQL>(stk, token.span).await?;
+						//HACK: This is an annoying hack to have geometries work.
+						//
+						// Geometries look exactly like objects and are a strict subsect of objects.
+						// However in code they are distinct and in surrealql the have different
+						// behavior.
+						//
+						// Geom functions don't work with objects and vice-versa.
+						//
+						// The previous parse automatically converted an object to geometry if it
+						// found an matching object. Now it no longer does that and relies on
+						// the 'planning' stage to convert it. But here we still need to do it
+						// in the parser.
+						if let Some(geom) = PublicGeometry::try_from_object(&object) {
+							PublicValue::Geometry(geom)
+						} else {
+							PublicValue::Object(object)
+						}
+					}
+					t!("}") => {
+						// Single-element object: `{value}`
+						// We could parse this in SQON, but in SurrealQL this is a block statement.
+						// So we instead throw an error and require the user to add a trailing
+						// comma for a set.
+						unexpected!(
+							self,
+							token,
+							"Ambiguous set syntax. Use `{value,}` instead of `{value}`"
+						);
```

Comment:
This will make the error be ``Unexpected token `}` expected Ambiguous set syntax...``
Instead put the hint in the actual hint.
```suggestion
						unexpected!(
							self,
							token,
							"`,`",
							=> "Sets with a single value must have at least a single comma"
						);
```
The above suggestion would have the error ``Unexpected token `}` expected  `,` `` and then the `Sets with a single...` hint will be in the code snippet, after the `^^^` indicating where the error happens.